### ISSUE TITLE
Commenting out loss=None causes much higher GPU memory usage in bing_bert.

### ISSUE THREAD

**Issue by szhengac** [microsoft/DeepSpeed#547]
I found that if I comment out `loss = None` in https://github.com/microsoft/DeepSpeedExamples/blob/fa1d1a71c48623db8a091d9cf636a5fe3b8f43c7/bing_bert/deepspeed_train.py#L176, the GPU memory usage increases significantly. It seems the GPU memory is accumulated with the next batch without manually dereferencing the object but I don't recall that we need this line in pure pytorch training code.
